src/transformers/quantizers/quantizer_quanto.py

```diff
@@ -0,0 +1,198 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import importlib
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+from packaging import version
+
+from .base import HfQuantizer
+from .quantizers_utils import get_module_from_name
+
+
+if TYPE_CHECKING:
+    from ..modeling_utils import PreTrainedModel
+
+from ..utils import is_quanto_available, is_torch_available, logging
+from ..utils.quantization_config import QuantoConfig
+
+
+if is_torch_available():
+    import torch
+
+logger = logging.get_logger(__name__)
+
+
+class QuantoHfQuantizer(HfQuantizer):
+    """
+    Quantizer for the quanto library
+    """
+
+    required_packages = ["quanto", "accelerate"]
+    requires_parameters_quantization = True
+    requires_calibration = False
+
+    def __init__(self, quantization_config: QuantoConfig, **kwargs):
+        super().__init__(quantization_config, **kwargs)
+        self.post_init()
+
+    def post_init(self):
+        r"""
+        Safety checker
+        """
+        if self.quantization_config.activations is not None and not self.pre_quantized:
+            raise ValueError(
+                "We don't support quantizing the activations with transformers library."
+                "Use quanto library for more complex use cases such as activations quantization, calibration and quantization aware training."
+            )
+
+    def validate_environment(self, *args, **kwargs):
+        if not is_quanto_available():
+            raise ImportError("Loading a quanto quantized model requires quanto library (`pip install quanto`)")
+        device_map = kwargs.get("device_map", None)
+        if device_map is not None and isinstance(device_map, dict):
+            if "cpu" in device_map.values() or "disk" in device_map.values():
+                if version.parse(importlib.metadata.version("accelerate")) <= version.parse("0.27.0"):
+                    raise ValueError(
+                        "You have a version of `accelerate` that is not compatible cpu/disk offload with quanto quantized model. "
+                        "You need to install a version of accelerate > 0.27.0."
+                    )
+
+    def update_torch_dtype(self, torch_dtype: "torch.dtype") -> "torch.dtype":
+        return torch_dtype
+
+    def update_weights_only_kwarg(self, weights_only_kwarg: Dict[str, Any]) -> Dict[str, Any]:
+        weights_only_kwarg["weights_only"] = False
+        return weights_only_kwarg
```

Comment:
let's document why we can't just load the weights ("because quanto relies on special structures") and also if there is .bin then we have safety dependency with quanto

Reply:
@SunMarc correct me if I am wrong but I think that is because when activations are quantized, quanto adds two buffers to quantized modules to store the input and output quantization scales (for weights also, but these are stored directly __inside__ the Tensor subclass with the same name, so it is transparent).
The serialized dict therefore contains more names.

Reply:
This is an issue compatibility with the safe serialization used by pytoch due to quanto special structure. You can check the issue I wrote [here](https://github.com/huggingface/quanto/issues/93) for more information ! 

@dacorvo, what you described is linked to the issue with the missing keys. I've added that comment to the related section. Thanks for explaining !